src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -82,7 +82,7 @@ public CorsResult EvaluatePolicy(HttpContext context, CorsPolicy policy)
         public virtual void EvaluateRequest(HttpContext context, CorsPolicy policy, CorsResult result)
         {
             var origin = context.Request.Headers[CorsConstants.Origin];
-            if (StringValues.IsNullOrEmpty(origin) || !policy.AllowAnyOrigin && !policy.Origins.Contains(origin))
+            if (StringValues.IsNullOrEmpty(origin) || !policy.AllowAnyOrigin && !policy.RegularOrigins.Any(r => r.IsMatch(origin)))
```

Comment:
Sure, anything with regex raises a red flag. We avoid them like the plague and performance results need to be shown whenever they are used. They also need a timeout so @blowdart doesn't freak out 😄